client-network/src/network_ops/default_network_ops_client.rs

```diff
@@ -492,9 +493,7 @@ where
                 )
             })?;
 
-        let signature = private_key
-            .sign(transaction.id())
-            .map(StakedStateOpWitness::new)?;
+        let signature = private_key.sign(&tx).map(StakedStateOpWitness::new)?;
```

Comment:
`private_key` is obtained on L486 with `self.wallet_client.private_key(...)` which returns the `PrivateKey`, not `PrivateKeyLocation`

Reply:
it may be worth adding a test with external "mock" HW wallet over TCP to check that the core transaction building functionality doesn't assume the a local private key